jvector-base/src/main/java/io/github/jbellis/jvector/graph/GraphIndexBuilder.java

```diff
@@ -225,18 +225,19 @@ public void cleanup() {
                     .average()
         ).join().orElse(Double.NaN);
 
+        // optimize entry node -- we do this before reconnecting, as otherwise, improving the entry node's
+        // connections will tend to disconnect any orphaned nodes reconnected to the entry node
+        updateEntryPoint();
```

Comment:
should we special case orphan reconnect to ignore entry node?  seems like it's important to not clutter that with poor quality edges